src/braket/default_simulator/simulator.py

```diff
@@ -379,26 +247,16 @@ def _observable_hash(observable: Observable) -> Union[str, Dict[int, str]]:
     @staticmethod
     def _generate_results(
         circuit_ir: Program,
-        non_observable_result_types: Dict[int, ResultType],
-        observable_result_types: Dict[int, ObservableResultType],
+        result_types: List[ResultType],
         simulation,
     ) -> List[ResultTypeValue]:
-
-        results = [0] * len(circuit_ir.results)
-
-        for index in non_observable_result_types:
-            results[index] = ResultTypeValue.construct(
+        return [
+            ResultTypeValue.construct(
                 type=circuit_ir.results[index],
-                value=non_observable_result_types[index].calculate(simulation),
+                value=result_types[index].calculate(simulation),
```

Comment:
Any idea why we separated `non_observable` from `observable` previously?

Reply:
The observable result types were extracted and their observables validated to ensure they were identical. Since that validation is no longer necessary, they don't have to be separated anymore. Honestly, they didn't even have to be split up so early even then, as the entire list could've just been filtered for the validation; doesn't matter now.